homeassistant/components/switch/rachio.py

```diff
@@ -0,0 +1,214 @@
+import logging
+import voluptuous as vol
+from datetime import timedelta
+
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util as util
+from homeassistant.components.switch import SwitchDevice, PLATFORM_SCHEMA
+from homeassistant.const import CONF_ACCESS_TOKEN
+
+REQUIREMENTS = ['https://github.com/Klikini/rachiopy'
+                '/archive/edf666de8ef3f9596ddc8d3a989e8c4b829a4319.zip'
+                '#rachiopy==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_RACHIO = 'rachio'
+
+CONF_MANUAL_RUN_MINS = 'manual_run_mins'
+manual_run_mins = 60
+
+MIN_UPDATE_INTERVAL = timedelta(minutes=5)
+MIN_FORCED_UPDATE_INTERVAL = timedelta(seconds=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ACCESS_TOKEN): cv.string,
+    vol.Optional(CONF_MANUAL_RUN_MINS): cv.positive_int
+})
+
+
+# Set up the component
+# noinspection PyUnusedLocal
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    global manual_run_mins
+
+    # Get options
+    manual_run_mins = config.get(CONF_MANUAL_RUN_MINS) or manual_run_mins
+    _LOGGER.debug("Rachio run time is " + str(manual_run_mins) + " min")
```

Comment:
This, among other things, is one of the many lint issues that CI is reporting that need to be fixed: https://travis-ci.org/home-assistant/home-assistant/jobs/233033184#L271